### PR TITLE
HPCC-17161 Add delete confirmation dialog

### PR DESCRIPTION
**Pull request by miguelvazq**
All details area should container confirmation dialog like the query areas do.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by richardkchapman** [comment]
Can you update the PR checklist please

**Comment by miguelvazq** [comment]
Yep,

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

**Review by GordonSmith — CHANGES_REQUESTED** [review]
We normally pass in the list of items that are about to be deleted?

**Comment by miguelvazq** [comment]
This is at the details level.

File: esp/src/eclwatch/LFDetailsWidget.js [hunk 1]
```diff
@@ -141,10 +141,12 @@ define([
...
         },
         _onDelete: function (event) {
-            this.logicalFile.doDelete({
-                load: function (response) {
-                }
-            });
+            if (confirm(this.i18n.YouAreAboutToDeleteThisFile)) {
+                this.logicalFile.doDelete({
+                    load: function (response) {
+                    }
```
Comment (GordonSmith) [inline esp/src/eclwatch/LFDetailsWidget.js]: Minor = Load callback not needed.

**Review by GordonSmith — APPROVED** [review]
Looks ok - one minor comment.

**Comment by miguelvazq** [comment]
@GordonSmith Minor comment updated.

**Comment by miguelvazq** [comment]
@rpastrana Can you review the Spanish translations?

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 2]
```diff
@@ -763,6 +763,9 @@ define(
     Year: "Año",
     YouAreAboutToDeleteBinding: "Está a punto de eliminar esta conexión. Desea continuar?",
     YouAreAboutToRemoveUserFrom: "Usted esta a punto de quitar un(os) usuario(s) de este grupo. ¿Desea continuar?",
+    YouAreAboutToDeleteThisFile: "Usted esta a punto de eliminar este archivo. ¿Desea continuar?",
+    YouAreAboutToDeleteThisQueryset: "Usted esta a punto de eliminar este grupo de query. ¿Desea continuar?",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/es/hpcc.js]: Looked through the file and for the most part we do translate query to consulta. Although I did see a few entries where we don't translate query.
And perhaps outside the scope of this issue, but there's a instace where we (prob me) misspelled Consulta:
ResetThisQuery: "Reajustar esta Consulat",

File: esp/src/eclwatch/nls/hpcc.js [hunk 3]
```diff
@@ -762,6 +762,9 @@ define({root:
     Year: "Year",
     YouAreAboutToRemoveUserFrom: "You are about to remove a user(s) from this group. Do you wish to continue?",
     YouAreAboutToDeleteBinding: "You are about to delete this binding. Are you sure you want to do this?",
+    YouAreAboutToDeleteThisFile: "You are about to delete this file. Are you sure you want to do this?",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/hpcc.js]: These didn't get translated

**Review by rpastrana — CHANGES_REQUESTED** [review]
@miguelvazq a couple of comments.

File: esp/src/eclwatch/nls/es/hpcc.js (hunk 2 above)
Comment (miguelvazq) [inline esp/src/eclwatch/nls/es/hpcc.js]: I will open a JIRA for this spelling mistake.

File: esp/src/eclwatch/nls/hpcc.js (hunk 3 above)
Comment (miguelvazq) [inline esp/src/eclwatch/nls/hpcc.js]: One is for the default English version and one for the Spanish file.

**Review by miguelvazq — COMMENTED** [review]
Changed 'query' to 'consulta'

**Review by rpastrana — APPROVED** [review]

**Comment by rpastrana** [comment]
@miguelvazq

**Comment by miguelvazq** [comment]
@richardkchapman This is ready to merge.

**Pull request merged**